apisix/discovery/eureka.lua

```diff
@@ -0,0 +1,247 @@
+        end
+        -- log.warn(method, url, body)
+        headers['Content-Type'] = 'application/json'
+    end
+
+    local httpc = http.new()
+    local timeout = local_conf.eureka.timeout
+    local connect_timeout = timeout and timeout.connect or 2000
+    local send_timeout = timeout and timeout.send or 2000
+    local read_timeout = timeout and timeout.read or 5000
+    httpc:set_timeouts(connect_timeout, send_timeout, read_timeout)
+    return httpc:request_uri(url, {
+        version = 1.1,
+        method = method,
+        headers = headers,
+        query = query,
+        body = body,
+        ssl_verify = false,
+    })
+end
+
+
+local function parse_instance(instance)
+    local status = instance.status
+    local overridden_status = instance.overriddenstatus or instance.overriddenStatus
+    if overridden_status and "UNKNOWN" ~= overridden_status then
+        status = overridden_status
+    end
+
+    if status ~= "UP" then
+        return
+    end
+    local port
+    if tostring(instance.port["@enabled"]) == "true" and instance.port["$"] then
+        port = instance.port["$"]
+        -- secure = false
+    end
+    if tostring(instance.securePort["@enabled"]) == "true" and instance.securePort["$"] then
+        port = instance.securePort["$"]
+        -- secure = true
+    end
+    local ip = instance.ipAddr
+    if not ipmatcher.parse_ipv4(ip) and
+            not ipmatcher.parse_ipv6(ip) then
+        log.error("invalid ip:", ip)
+        return
+    end
+    return ip, port, instance.metadata
+end
+
+
+local function fetch_full_registry(premature)
+    if premature then
+        return
+    end
+
+    local request_uri, basic_auth = service_info()
+    if not request_uri then
+        return
+    end
+
+    local res, err = request(request_uri, basic_auth, "GET", "apps")
+    if not res then
+        log.error("failed to fetch registry", err)
+        return
+    end
+
+    if not res.body or res.status ~= 200 then
+        log.error("failed to fetch registry, status = ", res.status)
+        return
+    end
+
+    local json_str = res.body
+    local data, err = core.json.decode(json_str)
+    if not data then
+        log.error("invalid response body: ", json_str, " err: ", err)
+        return
+    end
+    local apps = data.applications.application
+    local up_apps = core.table.new(0, #apps)
+    for _, app in ipairs(apps) do
+        for _, instance in ipairs(app.instance) do
+            local ip, port, metadata = parse_instance(instance)
+            if ip and port then
+                local nodes = up_apps[app.name]
+                if not nodes then
+                    nodes = core.table.new(#app.instance, 0)
+                    up_apps[app.name] = nodes
+                end
+                core.table.insert(nodes, {
+                    host = ip,
+                    port = port,
+                    weight = metadata and metadata.weight or default_weight,
+                    metadata = metadata,
```

Comment:
we need to remove `metadata` as discussed in mailing list.

Reply:
https://github.com/apache/incubator-apisix/pull/1440/files/2a2da03a7b71ad3314a89253fa478182fa66c22f#diff-98c27e6d94725a36de10557d23b1ca94R211

remove in here